Sources/MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -163,12 +213,14 @@ open class SimulatedLocationManager: NavigationLocationManager {
                                   verticalAccuracy: verticalAccuracy,
                                   course: newCoordinate.direction(to: lookAheadCoordinate).wrap(min: 0, max: 360),
                                   speed: currentSpeed,
-                                  timestamp: Date())
+                                  timestamp: getNextDate())
 
         self.simulatedLocation = location
 
         delegate?.locationManager?(self, didUpdateLocations: [location])
-        currentDistance = calculateCurrentDistance(currentDistance)
+        currentDistance += polyline.distance(to: newCoordinate) ?? 0
```

Comment:
Haven't you already calculated this distance?
```suggestion
        currentDistance += indexedNewCoordinate.distance
```